src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupSDCZeros.java

```diff
@@ -848,23 +848,52 @@ else if(points[c].o < of)
 		while(c < points.length && points[c].o < last)
 			c++;
 
-		c = processRow(points, sr, nCol, c, of, _data.getIndex(it.getDataIndex()));
+		c = processRowSparse(points, sr, nCol, c, of, _data.getIndex(it.getDataIndex()));
 
 	}
 
 	@Override
 	protected void denseSelection(MatrixBlock selection, P[] points, MatrixBlock ret, int rl, int ru) {
-		throw new NotImplementedException();
+				final DenseBlock dr = ret.getDenseBlock();
```

Comment:
Indentation

Reply:
oh yeah, comes from the + signs, the first block I edited in the IDE and + sign confused the IDE to make these wild indentations. After that I edited the block in a separate editor first^^